contracts/fio.address/fio.address.cpp

```diff
@@ -1735,8 +1732,45 @@ namespace fioio {
             }
             else check(false, "Failed to decrement eligible bundle counter"); // required to fail the parent transaction
         }
+
+        [[eosio::action]]
+        void xferescrow(const string &fio_domain, const string &public_key, const name &actor){
+            require_auth(EscrowContract);
+
+            FioAddress fa;
+            getFioAddressStruct(fio_domain, fa);
+
+            register_errors(fa, true);
+            fio_400_assert(isPubKeyValid(public_key), "public_key", public_key,
+                           "Invalid FIO Public Key", ErrorChainAddressEmpty);
+
+            auto domainsbyname = domains.get_index<"byname"_n>();
+            auto domains_iter = domainsbyname.find(string_to_uint128_hash(fio_domain));
+            fio_400_assert(domains_iter != domainsbyname.end(), "fio_domain", fio_domain,
+                           "FIO Domain not registered", ErrorDomainNotRegistered);
+
+            const uint32_t domain_expiration = domains_iter->expiration;
+            const uint32_t present_time = now();
+            fio_400_assert(present_time <= domain_expiration, "fio_domain", fio_domain, "FIO Domain expired. Renew first.",
+                           ErrorDomainExpired);
+
+            //Transfer the domain
+            string owner_account;
+            key_to_account(public_key, owner_account);
+            const name nm = name(owner_account);
+            domainsbyname.modify(domains_iter, _self, [&](struct domain &a) {
+                a.account = nm.value;
+            });
+
+            const string response_string = string("{\"status\": \"OK\"}");
+
+            fio_400_assert(transaction_size() <= MAX_TRX_SIZE, "transaction_size", std::to_string(transaction_size()),
+                           "Transaction is too large", ErrorTransaction);
+
+            send_response(response_string.c_str());
+        }
```

Comment:
Code block not needed since responses is handled in parent action.

Reply:
i realized that when looking over it again so I'll remove this